tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -224,6 +223,9 @@ def __init__(self, test_context, *args, **kwargs):
             # evict it frequently
             'cloud_storage_materialized_manifest_ttl_ms':
             self.params.cloud_storage_materialized_manifest_ttl_ms,
+
+            # disable spillover: it will be explicitly set in the tests when needed
+            'cloud_storage_spillover_manifest_size': None,
```

Comment:
again, probably i'm confused but this test tries to enable spillover by toggling max_segments rather than manifest_size

are you sure it does the right thing? https://github.com/redpanda-data/redpanda/blob/3d45a0750f83095c0d5ee0a7d50d3538734229c8/tests/rptest/infinite_retention/infinite_retention_test.py#L533